### PR TITLE
License NuCypher as GPLv3

### PR DESCRIPTION
**Pull request by tuxxy**
This software is free as in _freedom_.

### CODE REVIEW

File: cli/main.py [hunk 1]
```diff
@@ -1,3 +1,19 @@
+"""
```
Comment (michwill) [inline cli/main.py]: I respect the communism, but the comment should be after `#!/usr/bin/env python3`. Vim with flake8 should have told you that!

Comment (tuxxy) [reply cli/main.py]: I didn't use vim, I used a script to prepend the license header. If you see anymore, lmk.

File: examples/finnegans_wake_demo/finnegans-wake-concise-demo.py [hunk 2]
```diff
@@ -1,3 +1,19 @@
+"""
```
Comment (michwill) [inline examples/finnegans_wake_demo/finnegans-wake-concise-demo.py]: Is it really necessary to have this (huge!) disclaimer even in the demo script?

I know, RMS does this, but still...

Comment (jMyles) [reply examples/finnegans_wake_demo/finnegans-wake-concise-demo.py]: I'm absolutely in favor of removing it.

**Comment by mswilkison** [comment]
Under GPL, if someone forks Ursula into Forkula, and connects to the network they are not required to open source the code. With AGPL, they would be. Is my understanding correct?

Since AGPL is GPL for code that runs primarily over a network should we be using that?

**Comment by michwill** [comment]
Well, that is right. But seriously: imagine someone connecting to the network using a proprietary and perhaps malicious Ursula breaking all possible licenses. But you don't even know who that is.
How would AGPL help you in this case?

In the very worst case, we are able to hard fork and ban some Ursula if everyone agrees that this Ursula is malicious (and thus will follow the new fork).

So, I think, AGPL is a bit of an overkill here. GPL is helpful mainly dealing with largeco clients who wouldn't use the public network (so we force them to not ride for free), and I think just GPL seems enough for that.

Besides, as another data point, go-ethereum uses LGPL3, parity is GPL3

File: examples/finnegans_wake_demo/finnegans-wake-concise-demo.py (hunk 2 above)
Comment (KPrasch) [reply examples/finnegans_wake_demo/finnegans-wake-concise-demo.py]: I like the formality in some cases.. but not this one.  Im in favor of removing it.

File: cli/main.py (hunk 1 above)
Comment (KPrasch) [reply cli/main.py]: Just to nit-pick, I think this looks better / more convention with two new lines after the license text.

**Pull request merged**